src/main/java/seedu/address/model/iteration/UniqueIterationList.java

```diff
@@ -0,0 +1,142 @@
+package seedu.address.model.iteration;
+
+import static java.util.Objects.requireNonNull;
+import static seedu.address.commons.util.CollectionUtil.requireAllNonNull;
+
+import java.util.Iterator;
+import java.util.List;
+
+import javafx.collections.FXCollections;
+import javafx.collections.ObservableList;
+import seedu.address.model.commission.exceptions.DuplicateCommissionException;
+import seedu.address.model.iteration.exceptions.DuplicateIterationException;
+import seedu.address.model.iteration.exceptions.IterationNotFoundException;
+
+/**
+ * A list of iterations that enforces uniqueness between its elements and does not allow nulls.
+ * An iteration is considered unique by comparing using {@code Iteration#isSameIteration(Iteration)}. As such, adding
+ * and updating of iterations uses Iteration#isSameIteration(Iteration) for equality so as to ensure that the
+ * iteration being added or updated is unique in terms of identity in the UniqueIterationList. However, the removal of
+ * an iteration uses Iteration#equals(Object) so as to ensure that the iteration with exactly the same fields will be
+ * removed.
+ *
+ * Supports a minimal set of list operations.
+ *
+ * @see Iteration#isSameIteration(Iteration)
+ */
+public class UniqueIterationList implements Iterable<Iteration> {
+
+    private final ObservableList<Iteration> internalList = FXCollections.observableArrayList();
+    private final ObservableList<Iteration> internalUnmodifiableList =
+            FXCollections.unmodifiableObservableList(internalList);
+
+    /**
+     * Returns true if the list contains an equivalent iteration as the given argument.
+     */
+    public boolean contains(Iteration toCheck) {
+        requireNonNull(toCheck);
+        return internalList.stream().anyMatch(toCheck::isSameIteration);
+    }
+
+    /**
+     * Adds an iteration to the list.
+     * The iteration must not already exist in the list.
+     */
+    public void add(Iteration toAdd) {
+        requireNonNull(toAdd);
+        if (contains(toAdd)) {
+            throw new DuplicateIterationException();
```

Comment:
Currently since hasIteration is used to check for duplicate before addIteration is invoked in AddIterationCommand.execute() will this exception ever be thrown or its a fail safe?